src/psyclone/tests/domain/lfric/dofkern_test.py

```diff
@@ -142,3 +154,157 @@ def test_dof_kernel_invalid_field_vector():
+
+    '''
+    # Set up annexed dofs
+    config = Config.get()
+    lfric_config = config.api_conf("lfric")
+    monkeypatch.setattr(lfric_config, "_compute_annexed_dofs", annexed)
+
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "1.14_single_invoke_dofs.f90"),
+                           api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=dist_mem).create(invoke_info)
+    code = str(psy.gen)
+
+    # Distributed memory
+    if annexed and dist_mem:
+        expected = ("      loop0_start = 1\n"
+                    "      loop0_stop = f1_proxy%vspace%get_last_dof_annexed()"
+                    )
+    elif not annexed and dist_mem:
+        expected = ("      loop0_start = 1\n"
+                    "      loop0_stop = f1_proxy%vspace%get_last_dof_owned()"
+                    )
+
+    # Shared memory
+    elif not annexed and not dist_mem or \
+            annexed and not dist_mem:
+        expected = ("      loop0_start = 1\n"
+                    "      loop0_stop = undf"
+                    )
+
+    assert expected in code
+
+
+def test_indexed_field_args():
+    '''
+    Checks that the correct array references are generated for all field
+    arguments in a dof kernel. The index should be the same as the loop
+    index - 'df'.
+
+    '''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "1.14_single_invoke_dofs.f90"),
+                           api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(invoke_info)
+    code = str(psy.gen)
+
+    expected = ("CALL testkern_dofs_code(f1_data(df), f2_data(df), "
+                "f3_data(df), f4_data(df), scalar_arg, undf)")
+
+    assert expected in code
+
+
+def test_undf_initialisation():
+    '''
+    Test that the 'bare' undf name is used when intialising the undf
+    variable for dof kernels.
+
+    '''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "1.14_single_invoke_dofs.f90"),
+                           api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(invoke_info)
+    code = str(psy.gen)
+
+    declaration = "INTEGER(KIND=i_def) undf"
+    initalisation = "undf = f1_proxy%vspace%get_undf()"
+
+    assert declaration in code
+    assert initalisation in code
+
+
+def test_function_space_bare_undf():
+    '''
+    Test that the correct undf name ("undf") is stored in DynFunctionSpaces
+    list of undf_names when a kernel is found to operate on 'dof'
+
+    '''
+    _, invoke = get_invoke(os.path.join(BASE_PATH,
+                                        "1.14_single_invoke_dofs.f90"),
+                           TEST_API,
+                           idx=0, dist_mem=False)
+    schedule = invoke.schedule
+    kernel = schedule.walk(LFRicKern)[0]
+    test_fs = DynFunctionSpaces(kernel)
+
+    assert 'undf' in test_fs._var_list
+
+
+def test_compiles(tmpdir):
```

Comment:
You don't need this new test. Just add the bit that tests compilation at the end of the existing tests.

Reply:
ahh, thank you

Reply:
This test can go now.